### ISSUE TITLE
Typescript: Configuration variables shouldn't be const

### ISSUE THREAD

**Issue by ColonelThirtyTwo** [bigskysoftware/htmx#2850]
In `htmx.d.ts`, the variables in the `htmx.config` are declared using `const`, so projects using typescript will be unable to set them without typescript throwing an error.

Same with `htmx.logger`.

**Comment by Telroshan** [comment]
Oof, good catch, hadn't realized this!

Ok so it's hard to get `tsc` to generate what we want, we had to do crazy tricks sometimes to get a proper result (see #2336, #2653)
If you figure out a way to have tsc (see the [`npm run type-declarations`](https://github.com/bigskysoftware/htmx/blob/master/package.json#L34) command) generate the correct thing, a bugfix PR would be very much welcome!

**Comment by ColonelThirtyTwo** [comment]
@Telroshan `type-declarations` fails because it can't find `tsconfig.json` - I don't see one anywhere in the repo.

**Comment by ColonelThirtyTwo** [comment]
Looks like e1143de8500dc91f54ed40aa05306a0049720804 generates the `d.ts` as part of `npm run types-generate`, and both the `type-declarations` command and the `src/html.d.ts` file are not used anymore. Should probably delete those.

**Comment by ColonelThirtyTwo** [comment]
Looks like Typescript v5.1 and up generate `let` instead of `const` for all of the exported members of the htmx namespace, so the solution might just be to update typescript.

**Comment by MichaelWest22** [comment]
Yeah newer typescript generates with let which hopefully solves your issue.  How do we test the updated esm.d.ts file to confirm it still works fine and resolves the problem?

v5.5 version seems to add additional changes because it has stricter rules it seems.  I don't know if the updates type inference it does is an improvement we want now or not.
example:
```javascript
// typescript v4.9.5
    const trigger: (elt: string | EventTarget, eventName: string, detail?: any) => boolean;
    const ajax: (verb: HttpVerb, path: string, context: string | Element | HtmxAjaxHelperContext) => Promise<void>;
    const find: (eltOrSelector: string | ParentNode, selector?: string) => Element;
    const findAll: (eltOrSelector: string | ParentNode, selector?: string) => NodeListOf<Element>;
    const closest: (elt: string | Element, selector: string) => Element;

// typescript v5.4.5
    let trigger: (elt: string | EventTarget, eventName: string, detail?: any) => boolean;
    let ajax: (verb: HttpVerb, path: string, context: string | Element | HtmxAjaxHelperContext) => Promise<void>;
    let find: (eltOrSelector: string | ParentNode, selector?: string) => Element;
    let findAll: (eltOrSelector: string | ParentNode, selector?: string) => NodeListOf<Element>;
    let closest: (elt: string | Element, selector: string) => Element;

// typescript 5.5.4
    let trigger: (elt: EventTarget | string, eventName: string, detail?: any | undefined) => boolean;
    let ajax: (verb: HttpVerb, path: string, context: Element | string | HtmxAjaxHelperContext) => Promise<void>;
    let find: (eltOrSelector: ParentNode | string, selector?: string) => Element | null;
    let findAll: (eltOrSelector: ParentNode | string, selector?: string) => NodeListOf<Element>;
    let closest: (elt: Element | string, selector: string) => Element | null;
```

**Comment by MichaelWest22** [comment]
reviewed all the changes made with v5.5.4 and they all seem to better reflect the jsdoc annotations so update my PR to this version.

**Comment by dkristia** [comment]
Shouldn't this issue be closed now that #2853 has been merged?

**Comment by MichaelWest22** [comment]
@ColonelThirtyTwo fix now in v2.0.3 so please retest and close if it is now resolved

**Comment by 1cg** [comment]
yep, fixed thank you everyone
